server/jetstream_cluster.go

```diff
@@ -5043,6 +5043,11 @@ func (js *jetStream) applyConsumerEntries(o *consumer, ce *CommittedEntry, isLea
 				o.mu.Lock()
 				err = o.store.UpdateDelivered(dseq, sseq, dc, ts)
 				o.ldt = time.Now()
+				// Need to send message to the client, since we have quorum to do so now.
+				if pmsg, ok := o.pendingDeliveries[sseq]; ok {
+					o.outq.send(pmsg)
+					delete(o.pendingDeliveries, sseq)
```

Comment:
We may want to check `len(o.pendingDeliveries)` vs `cap(o.pendingDeliveries)` here to ensure that we shrink & repopulate the map if the usage pattern changes, otherwise we could hold onto memory until the leader changes. Maybe if the length drops to below one quarter of the capacity or so. WDYT?

Reply:
> here to ensure that we shrink & repopulate the map if the usage pattern changes

Good point, we should at least do that check when `MaxAckPending` changes. As that should probably shrink it.

Am a bit wary of constantly shrinking and resizing the map. When you do fetch N, where N<MaxAckPending, the `o.pendingDeliveries` will be max N in size. Then all messages will be sent to the client, and it could technically shrink. But if the client then fetches again, we'll need to allocate again. Hence why I'm thinking about only resizing based on `MaxAckPending` being lowered.

Also, not entirely sure, but believe `cap(..)` can't be used on a map?

Reply:
Ah, indeed it can't, well let's do so on `MaxAckPending` updates at least and we can watch how it goes.

Reply:
Done, it now gets reset if `MaxAckPending` gets lowered.